### ISSUE TITLE
Corporation tuning needed.

### ISSUE THREAD

**Issue by kedearian** [danielyxie/bitburner#2350]
As per discord: 

There are currently corps are in a very unbalanced state, with gameplay that trivializes the need for any hacking, scripting, or other mechanics. You are able to get massive evaluations by gaming your income to be able to hyper-scale your corp and beat almost any node in less than 6 hours with zero other mechanics used. 

Steps to reproduce 

1. Start software
2. Buy smart supply
3. get all 6 cities with warehouses for 6x production multi
4. get 900 storage in each city
5. horde AI limiters (set sell to 0 at MP so you hold them all)
6. place works as 1 operation / 1 engineer / 1 manager / 3 research
7. when storage fills, quickly flip all AI limiters to sell MAX at MP
8. when income skyrockets, get a high evaluation.
repeat this process (max storage, stockpile, sell off in 1 market tick) to get a second huge bump of money. After that you can play normally (boost averts/production/worker counts). Since you get a huge influx of cash even keeping 10% of your total company becomes viable with how fast you scale.

My suggestion is setting the investor to take an average of your income over the last 1-5 minutes to help level out this kind of play, but there are probably other ways that would work and not completely destroy the clever use of mechanics.

**Comment by Janaszar** [comment]
Some differences with my method:
Wait until you get some product developed (usually software, but have done food).  It has a higher profit ratio and market sell price (~25k for software last I checked, AI cores are ~14k), as well as being able to use the "Set price for all cities" function to make the whole process easier.
Occasionally the amount in storage is so high, that even at MP, you have more than one cycle to sell.  As such, I would second the idea that the investor amount is averaged over 1~5 minutes.

Other things:
Awareness, Popularity, and the Advertising Multiplier is often able to feedback very quickly once you hit about 1q/s corp income.
Wilson apparently increases the AdVert.Inc multipliers by +0.5%, making them very strong as you will be doing a +20% each buy with only ~40 upgrades of it.
In line with this, there should probably be a sanity check on the Awareness/Popularity values, as they can overflow around e300 (e308?)

A lot of the mechanics are obfuscated.  I was working on developing an Excel spreadsheet to try and plan corp stuff, and the things like having each employee's stats play a roll just made things weird when expanding over a handful of people. Other areas where multiple people that have important stats has the stats displayed obviously (gangs, sleeves), or if the people are just "generic fodder", they're just a number (BladeBurners).  Same sort of issue with their bonuses/upgrades; I see some result in the end, but I couldn't really tell how useful having high morale/happiness/energy is, or what the upgrades (FocusWires, for example) do for my income.
Demand and Competition is similar; I thought it was just static variables set when developing a product, but just looking now, they move over time for products to make them useless after some time.  I have no idea how long it would take to make a product useless, but that's just not covered anywhere outside the code, and you need to buy two 5b upgrades to notice that.

I feel like if the APIs were cheaper (1TB is pretty much "you're ready to finish" in my experience, but also the cost of 50b each means they're later upgrades if you go for them), they could be useful early on.  That in turn, could allow players to use Corps more like Sleeves/Gangs, where a simple automation script can get you to a reasonably good place in reasonable time.

Some other thoughts:
Haven't tried it, but if the Export materials function allowed you to send your own manufactured material somewhere else to be used as a source for something else you make, it could be useful in replacing advertising.  Have your farm make food, which goes into your food industry, which produces food-products. By having the agriculture industry produce higher quality items, your food industry uses higher quality items, thus providing a higher value to the items sold.  This gives a reason to expand into other industries (which, I see no point to now, other than to open up something like Real Estate which seems like a huge money maker, but you can't start with it).
As such, turning the Corps from "develop one very well for nigh infinite money" to "develop several/all for great returns" or "develop and chain them all for even better returns" becomes a nice idea/goal, and allows automation to be more useful.

I think I'm rambling a bit here... Hope this helps.

**Comment by pigalot** [comment]
The core issue with corps is the cost scaling, its so out of proportion with what can be gained normally that it means if you mess up once you cannot correct it for days.

on the subject of what corps give if corps didn't give infinite money they wouldn't really have a reason to exist any more, they would just be much worse gangs. So they would need to give something at least as useful or why would you bother with them.

**Comment by phyzical** [comment]
hmm one quick idea.. 
* dividends get taxed much more heavily
* but can buy augmentations at some other scale? maybe 5 purchases ahead of current scale found in the normal approach to buy augmentations

some farfetched ideas
* maybe purchase bonus time? but then that could be used to self boost a corp for infinite bonus time, maybe if bonus time could be purchased for specific mechanics rather than everything?
* purchase "guides"?
* purchase contract solutions?
* purchase ascii art 😆 and make a "art gallery" maybe as a "collectors" sort of thing 

but besides that im not too sure, we already can "buy favor" what else could we do at this point?

**Comment by rj45** [comment]
I would like to bump this.

I don't feel like investment fraud was intended, and a relatively simple fix is just to make valuations look at the average income over several ticks. That would make spending 30 minutes waiting for warehouses to fill with zero income completely pointless as it would tank the valuation, and ruin the investment offer. Maybe you can still game it a bit by adjusting to stockpile more slowly but keep positive income, but you wouldn't be able to get nearly as ridiculously lucrative investment offers doing this.

Currently by using the above outlined method I can complete an easy node in 5-6 hours. It takes around 3 or 4 hours to get 150b using hacking, stonks, the gift and hacknet servers all working in concert. Then once I have corps, investment fraud initially takes 30 minutes to fill a warehouse, but then I get a 5-10t investment round from it. That is enough to get enough employees to crank out my first product pretty fast. Once I have my first product, if I do investment fraud again I can get a 50 or 100t investment. That's enough to get the second product very fast, plus a lot of wilson analytics and adverts. I can easily complete the node in 1-2 hours at that point.

Even at an oppressive 0.4 soft cap in BN 12.45 (ish), I can complete the node in roughly 12 hours using the above method. It takes more like 8 hours to get to 150b, but once I have it, investment fraud is such a huge spring board that I am done the node very shortly after.

I am happy to do some science and do some BN1 runs comparing with investment fraud and without it. I am also happy to try to set up a PR.

**Comment by phyzical** [comment]
@rj45 sure! if we can nerf the valuation im pretty sure hydro would accept

**Comment by pigalot** [comment]
Frankly without making corps something useful this will just nerf them to the point where there is 0 point using them.

All corps have going for them is they give a lot of money very quickly without that they are a broken mess of an unfun mechanic that is much worse that just booting a gang up.

**Comment by rj45** [comment]
I disagree. They can win most nodes in half to a third the time it takes to win with gangs. So I rarely reach for gangs anymore. Having them be much closer I think would be a good thing. I don't think it's very fun when you have to choose one mechanic or you're just wasting time. Having a choice between two or even three mechanics that are similar in time-to-win I think would be very good.

But I can run some experiments to see how much of an effect nerfing investment fraud will have. It will take a couple days to run those experiments. Stay tuned.

**Comment by pigalot** [comment]
You might be correct if corps were not the broken mess they are at the moment, if corps were fun in its own right you would have a point.

**Comment by rj45** [comment]
This is my very late game save, 404 int, all the nodes completed (except 6 and 7), 48 levels of BN 12, running the test in BN1:

- 1h15m: got 150b and corp started (using the gift and crime to boost hacknet servers, then the gift and hacknet servers to boost hacking and stonks)
- 1h53m: investment fraud got me a 10.5t first investment offer just selling AI cores, before first product
- 2h42m: investment fraud with the first finished product got 234t second investment offer
- 3h32m: investment fraud with the first three products got 205q third investment offer
- 3h37m: my scripts started using corp funds to bribe factions with hacking income to pay for augs
- 3h50m: my script starts trying to join daedalus
- 3h52m: red pill bought
- 3h58m: back to the bitverse

There's some obvious inefficiencies in my scripts, but my scripts finished the game before my corp even got the fourth investment and went public. But I think that furthers my point about how OP investment fraud is.

This next run is the same setup but I just commented out the investment fraud code and just took the offer, otherwise the code is identical.

- 1h15m: 150b gotten and corp created
- 1h26m: full happiness gotten and first investment offer accepted for 39.7b
- 3h32m: round two investment accepted after the first product finished for 95.4b
- 5h47m: my progression script bugged out here and got stuck, it was at the 2nd daedalus aug buy
- 6h06m: round three investment accepted after product 3 finished for 503b
- 9h08m: round four investment accepted after product 6 finished for 1.48t
- 11h26m: I wake up to see it had gotten stuck at 5h47m, give it a manual nudge 
- 11h32m: red pill bought from hacking alone, no help from corp
- 11h35m: I manually went public with 50m shares, got 237b in the IPO
- 11h40m: I should have been done the node by now

Basically I now won the node by hacking alone. But keep in mind that I have done zero optimization of my corp script for this case. I basically stopped trying to improve it once I could win the node so quickly using investment fraud. I had no incentive to try to optimize my production modifiers, train employees, expand into more divisions, etc because I could get crazy money very quickly using investment fraud. There's also using hacknet to pump up my corp in the early game and I have never felt a need to do that either.

**TL;DR**

I finished BN1 in 4 hours with investment fraud. Without investment fraud I would have finished BN1 in 6 hours with just hacking alone and no help from my corp.

Another key insight: with investment fraud I have no incentive to explore other mechanics of corps, the money I get is crazy good without investing any more time. Without investment fraud I would actually have a reason to explore other divisions, employee training and optimization, production modifier optimization, etc in order to get higher investment rounds. I would find that a lot more fun than just having winsauce.

**Comment by PSEUDOSTAGE** [comment]
I do not see any particularly strong reason to believe that even outright removing investment fraud would alone encourage diversifying engagement within corporations into other industries. The "correct" ways to do corps would remain the "correct" ways to do corps, what works would still work, just slower.

This is a nerf to corporations on the whole under the assumption that the mechanic is too powerful, this is not a nerf to investment fraud. The same end effect could be achieved by "nerfing building sizes because being able to hire so many employees means the node can be finished quickly" or through "nerfing dividends by making you spend a significant amount of them continuously bribing municipal politicians because otherwise the node can be finished quickly". The point of this nerf has nothing specifically to do with investment fraud, the point of the nerf is to make the reward for BN3 less rewarding.

I agree that having reasons to engage with more industries, training, exports, etc would be wonderful. This nerf does not provide those reasons.

**Comment by phyzical** [comment]
i thought this too at first, but the pr i think only hits the valuation not the dividends from a corp right? not sure this pr affects the overall performance of a corp?

**Comment by rj45** [comment]
I have updated my PR to make the amount of nerfing configurable in the constants. I wasn't expecting this change to be controversial, but since there's disagreement, instead of nerfing investment fraud to the point where it's not a viable mechanic anymore, we could just make it much less profitable but still plenty viable.

Currently as it stands, investment fraud gives you roughly 250x the investment offer you would get without it. I set the constant to 10, which would make investment fraud give around 25x -- still very viable and worthwhile to do, but a lot less over powered.

**Comment by pigalot** [comment]
Again you miss the point rj45, the only reason corps are used at all is the current power.

Corps are not an engaging mechanic all. this change will do is make corps something you look at once in bn3 then never again.

**Comment by rj45** [comment]
Okay, the results are in. With the current PR I got the following investment rounds using the same code as before:

1.125t
6.387t
52.704t
12.494q

The last investment round is enough to push me into trillions per second of corp profit, and took only 6 hours to reach rather than 4 hours. So it's a pretty mild nerf.... not sure it's even worth it. I would say the constant probably needs to be more like 15 or 20 to be effective.

And for the record I have been talking in Discord and it seems like several people agree with @pigalot. I don't really understand the argument, but I think differently than most people so I am willing to concede. I leave it to hydro to decide... feel free to close the PR or tweak the constant and merge it or whatever.... I am just going to drop this now.
